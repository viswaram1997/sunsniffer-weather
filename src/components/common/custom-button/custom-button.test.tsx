import { render, fireEvent } from "@testing-library/react";
import Button from "./index"
const myMock = jest.fn();

describe('Custom button', () => {

    test('should match snapshot', () => {
        const result = render(<Button
            label='Submit'
        />)
        expect(result.baseElement).toMatchSnapshot();
    })

    test('should display proper value', () => {
        const result = render(<Button
            label='Click Here'
        />)
        const label = result.findByText('Click Here')
        expect(label).not.toBeNull()
    })

    test('should call on click function', async () => {
        const result = render(<Button
            label='Click Here'
            onClick={myMock}
        />)
        const button = await result.findByTestId('custom-button')
        fireEvent.click(button)
        expect(myMock).toBeCalled()
    })

    test('should have button type by default', async () => {
        const result = render(<Button
            label='Click Here'
            onClick={myMock}
        />)
        const button:any = await result.findByTestId('custom-button')
        expect(button.type).toEqual('button')
    })

});