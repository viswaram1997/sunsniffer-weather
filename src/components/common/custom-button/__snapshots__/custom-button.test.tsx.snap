// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom button should match snapshot 1`] = `
<body>
  <div>
    <button
      class="btn btn-primary"
      data-testid="custom-button"
      type="button"
    >
      Submit
    </button>
  </div>
</body>
`;
